src/components/ExpansionPanel/index.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isBoolean from 'lodash/isBoolean';
+import isEmpty from 'lodash/isEmpty';
+
+import Collapse from '../Collapse';
+import Header from './Header';
+import Body from './Body';
+
+const ExpansionPanel = ({
+  children,
+  collapseProps,
+  defaultExpanded,
+  expanded: expandedProp,
+  onToggle,
+  ...passedProps
+}) => {
+  const [uncontrolledExpanded, setUncontrolledExpanded] = useState(
+    defaultExpanded
+  );
+  const isControlled = !isBoolean(defaultExpanded);
+  const expanded = isControlled ? expandedProp : uncontrolledExpanded;
+  const [header, ...body] = React.Children.toArray(children);
+
+  const handleToggle = event => {
+    if (!isControlled) {
+      setUncontrolledExpanded(!expanded);
+    }
+    if (onToggle) {
+      onToggle(event, !expanded);
+    }
+  };
+
+  return (
+    <div {...passedProps}>
+      {React.cloneElement(header, { expanded, onClick: handleToggle })}
+      <Collapse isOpen={expanded} {...collapseProps}>
+        {body}
+      </Collapse>
+    </div>
+  );
+};
+
+ExpansionPanel.propTypes = {
+  /**
+   * The content of the expansion panel.
+   */
+  children: props => {
+    const children = React.Children.toArray(props.children);
+    if (isEmpty(children)) {
+      return new Error('ExpansionPanel: children is required.');
+    }
+    if (children[0].type !== Header) {
```

Comment:
I like this very much, but test it carefully. iirc, there used to be issues where hot reloader or something in the build process would stop this simple type check from working.

The old recommendation was to add a custom static type to the component, like `Header.displayName = "XXX.Header"`, and check that.

However, this way is better, if it works. Maybe this was something that's changed in recent versions.

Reply:
not sure about this

Reply:
If it's working, this is fine.